app/src/main/java/com/pokerio/app/utils/GameState.kt

```diff
@@ -303,38 +305,41 @@ object GameState {
     }
 
     fun removePlayer(playerHash: String, newAdmin: String? = null) {
-        // Check if this player is being removed
-        val isThisPlayerRemoved = players.find {
-            sha256(it.playerID) == playerHash
-        } != null
+        if (!isInGame()) {
+            return
+        }
+
+        val isThisPlayerRemoved = players.find { sha256(it.playerID) == playerHash } != null
+
         if (isThisPlayerRemoved) {
             resetGameState()
-        } else {
-            val removedPlayer = players.find { it.playerID == playerHash }
-                ?: throw Exception("Player to be removed not found")
-
-            if (removedPlayer.isAdmin && newAdmin == null) {
-                throw Exception("Admin removed without new admin given")
-            } else if (removedPlayer.isAdmin) {
-                val isThisPlayerNewAdmin = players.find {
-                    sha256(it.playerID) == newAdmin
-                } != null
-
-                if (isThisPlayerNewAdmin) {
-                    isPlayerAdmin = true
-                    val thisPlayer = players.find {
-                        sha256(it.playerID) == newAdmin
-                    }
-                    thisPlayer!!.isAdmin = true
-                } else {
-                    val newAdminPlayer = players.find { it.playerID == newAdmin }
-                    newAdminPlayer!!.isAdmin = true
-                }
-            }
+            return
+        }
+
+        val removedPlayer = players.find { it.playerID == playerHash }
+            ?: throw Exception("Player to be removed not found")
 
-            players.removeIf { it.playerID == playerHash }
-            playerRemovedCallbacks.forEach { it.value(removedPlayer) }
+        if (removedPlayer.isAdmin && newAdmin == null) {
+            throw Exception("Admin removed, but new admin was not set")
+        }
+
+        if (removedPlayer.isAdmin) {
+            val isThisPlayerNewAdmin = players.find { sha256(it.playerID) == newAdmin } != null
```

Comment:
You are doing the same find here and in line 330, can't you find the player and simply check if  he is not null in 'if'?

Reply:
I removed this as well as improving the actual removal of the player (line 336)